plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +215,395 @@ in
+      get_element_icon = defaultNullOpts.mkLuaFn null ''
         Lua function returning an element icon.
 
         ```
         fun(opts: IconFetcherOpts): string?, string?
         ```
       '';
 
-      showCloseIcon = helpers.defaultNullOpts.mkBool true "Whether to show the close icon.";
+      show_close_icon = defaultNullOpts.mkBool true ''
+        Whether to show the close icon.
+      '';
 
-      showTabIndicators = helpers.defaultNullOpts.mkBool true "Whether to show the tab indicators.";
+      show_tab_indicators = defaultNullOpts.mkBool true ''
+        Whether to show the tab indicators.
+      '';
 
-      showDuplicatePrefix = helpers.defaultNullOpts.mkBool true "Whether to show the prefix of duplicated files.";
+      show_duplicate_prefix = defaultNullOpts.mkBool true ''
+        Whether to show the prefix of duplicated files.
+      '';
 
-      enforceRegularTabs = helpers.defaultNullOpts.mkBool false "Whether to enforce regular tabs.";
+      duplicates_across_groups = defaultNullOpts.mkBool true ''
+        Whether to consider duplicate paths in different groups as duplicates.
+      '';
+
+      enforce_regular_tabs = defaultNullOpts.mkBool false ''
+        Whether to enforce regular tabs.
+      '';
 
-      alwaysShowBufferline = helpers.defaultNullOpts.mkBool true "Whether to always show the bufferline.";
+      always_show_bufferline = defaultNullOpts.mkBool true ''
+        Whether to always show the bufferline.
+      '';
 
-      persistBufferSort = helpers.defaultNullOpts.mkBool true "Whether to make the buffer sort persistent.";
+      auto_toggle_bufferline = defaultNullOpts.mkBool true ''
+        Whether to automatically toggle bufferline.
+      '';
 
-      maxPrefixLength = helpers.defaultNullOpts.mkInt 15 "Maximum prefix length";
+      persist_buffer_sort = defaultNullOpts.mkBool true ''
+        Whether to make the buffer sort persistent.
+      '';
 
-      sortBy = helpers.defaultNullOpts.mkStr "id" "sort by";
+      move_wraps_at_ends = defaultNullOpts.mkBool true ''
+        Whether or not the move command "wraps" at the first or last position.
+      '';
 
-      diagnostics = helpers.defaultNullOpts.mkNullable (
-        with types;
-        either bool (enum [
-          "nvim_lsp"
-          "coc"
-        ])
-      ) false "diagnostics";
+      max_prefix_length = defaultNullOpts.mkInt 15 ''
+        Maximum prefix length used when a buffer is de-duplicated.
+      '';
 
-      diagnosticsIndicator = helpers.defaultNullOpts.mkLuaFn null "Either `null` or a function that returns the diagnostics indicator.";
+      sort_by = defaultNullOpts.mkStr "id" ''
+        How to sort the buffers.
 
-      diagnosticsUpdateInInsert = helpers.defaultNullOpts.mkBool true "Whether diagnostics should update in insert mode";
+        Possible values: `insert_after_current`, `insert_at_end`, `id`, `extension`, `relative_directory`, `directory`, `tabs`.
```

Comment:
This looks a lot like an enum at first glance? Or is that a non-exhaustive list of examples?

Reply:
Ah yeah, I meant to come back and make this an enum, I believe. Thanks.